### PR TITLE
bugfix: retry multi times if failed to report pieces

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnopg@gmail.com>

### Ⅰ. Describe what this PR did

Sometimes dfget will fail to report pieces to supernode for some reason, such as supernode cannot handle it. And then supernode will mark that this task has not been completed forever. That's not expected. So we should add a retry mechanism to reduce the chance of errors.

### CODE REVIEW

File: dfget/core/downloader/p2p_downloader/client_writer.go [hunk 1]
```diff
@@ -202,12 +203,34 @@ func startSyncWriter(q queue.Queue) queue.Queue {
...
+	reportPieceRequest := &types.ReportPieceRequest{
 		TaskID:     piece.TaskID,
 		Cid:        cw.cfg.RV.Cid,
 		DstCid:     piece.DstCid,
 		PieceRange: piece.Range,
-	})
+	}
+
+	var retry = 0
+	var maxRetryTime = 3
```
Comment (lowzj) [inline dfget/core/downloader/p2p_downloader/client_writer.go]: The following codes will send request to supernode at most 4 times, is it expected?

File: dfget/core/downloader/p2p_downloader/client_writer.go [hunk 2]
```diff
@@ -202,12 +203,34 @@ func startSyncWriter(q queue.Queue) queue.Queue {
...
+				logrus.Warnf("success to report piece with request(%+v) after retrying (%d) times", reportPieceRequest, retry)
+			}
+			break
+		}
+		if retry >= maxRetryTime {
+			logrus.Errorf("failed to report piece to supernode with request(%+v) even after retrying max retry time", reportPieceRequest)
+			break
+		}
+
+		sleepTime := time.Duration(rand.Intn(1000)+500) * time.Millisecond
```
Comment (lowzj) [inline dfget/core/downloader/p2p_downloader/client_writer.go]: The time to sleep is a bit long. It's better to decrease the interval time of first retry. How about [50, 500]ms?

File: dfget/core/downloader/p2p_downloader/client_writer.go (hunk 1 above)
Comment (Starnop) [reply dfget/core/downloader/p2p_downloader/client_writer.go]: I don't think so. I use `>=` here `if retry >= maxRetryTime`.

File: dfget/core/downloader/p2p_downloader/client_writer.go (hunk 2 above)
Comment (Starnop) [reply dfget/core/downloader/p2p_downloader/client_writer.go]: DONE.

File: dfget/core/downloader/p2p_downloader/client_writer.go (hunk 1 above)
Comment (Starnop) [reply dfget/core/downloader/p2p_downloader/client_writer.go]: Updated.

**Pull request merged**